dev/PowerNotifications/PowerNotifications.idl

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+// REVIEW: We're not doing doing any versioning within IDL/WinMDs, right?
+
+namespace Microsoft.Windows.System.Power
+{
+    // Enums duplicated from IPowerManagerStatics
+    enum EnergySaverStatus
+    {
+        Uninitialized = 0,
+        Disabled,
+        Off,
+        On
+    };
+
+    enum BatteryStatus
+    {
+        NotPresent = 0,
+        Discharging,
+        Idle,
+        Charging
+    };
+
+    enum PowerSupplyStatus
+    {
+        NotPresent = 0,
+        Inadequate,
+        Adequate
+    };
+
+    // Enums based on on PowerSettingRegisterNotification's state
+    enum PowerSourceKind
+    {
+        AC = 0,
+        DC
+    };
+
+    enum DisplayStatus
+    {
+        Off = 0,
+        On,
+        Dimmed
+    };
+
+    enum EffectivePowerMode
+    {
+        // EFFECTIVE_POWER_MODE_V1 values
+        BatterySaver,
+        BetterBattery,
+        Balanced,
+        HighPerformance,
+        MaxPerformance,
+        // EFFECTIVE_POWER_MODE_V2 additional values
+        GameMode,
+        MixedReality
+    };
+
+    enum UserPresenceStatus
+    {
+        Present = 0,
+        Absent
+    };
+
+    enum SystemSuspendStatus
+    {
+        Uninitialized = 0,
+        Entering,
+        AutoResume,
+        ManualResume
+    };
+
+    static runtimeclass PowerManager
+    {
+        // Properties and events duplicated from IPowerManagerStatics.
+        static EnergySaverStatus EnergySaverStatus{ get; };
+        static event Windows.Foundation.EventHandler<Object> EnergySaverStatusChanged;
+
+        // TODO : Review API shape. We might want the BatteryStatusChanged event to
```

Comment:
Outstanding TODO. We answered this during design review: sending data with WinRT events is generally avoided because that data may very well be outdated by the time the event is processed (e.g. if the app receiving the event was suspended). Preferred to just signal the event, and have the event subscriber ask for the data when they process the event. Comment can be removed.